benchmark/primitives/context_test.go

```diff
@@ -139,7 +141,8 @@ func BenchmarkContextWithValue(b *testing.B) {
 		{"newContextWithGlobalKey", newContextWithGlobalKey},
 	}
 
-	pCtx := context.Background()
+	pCtx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
```

Comment:
Here, we are testing `context.WithValue()`, so it is totally OK to use the background context.

What you can fix here though are the `BenchmarkTimer*` functions. Those are using arbitrary large values, those should be changed use the `defaultTestTimeout`.